core/drivers/crypto/stm32/stm32_cryp.c

```diff
@@ -1229,6 +1233,51 @@ TEE_Result stm32_cryp_update(struct stm32_cryp_context *ctx, bool last_block,
 	return res;
 }
 
+static TEE_Result stm32_cryp_reset(void)
+{
+	TEE_Result ret = TEE_SUCCESS;
+
+	if (!cryp_pdata.reset)
+		return ret;
+
+	ret = rstctrl_assert_to(cryp_pdata.reset, TIMEOUT_US_1MS);
+	if (ret)
+		return ret;
+
+	udelay(CRYP_RESET_DELAY);
+
+	return rstctrl_deassert_to(cryp_pdata.reset, TIMEOUT_US_1MS);
+}
+
+static TEE_Result stm32_cryp_pm(enum pm_op op, uint32_t pm_hint,
+				const struct pm_callback_handle *hdl __unused)
+{
+	TEE_Result ret = TEE_ERROR_NOT_IMPLEMENTED;
+
+	switch (op) {
+	case PM_OP_SUSPEND:
+		clk_disable(cryp_pdata.clock);
+		ret = TEE_SUCCESS;
+		break;
+	case PM_OP_RESUME:
+		ret = clk_enable(cryp_pdata.clock);
+		if (ret)
```

Comment:
I think panicking also here makes sense, does it?

Reply:
Yes could be a good idea to panic, we expect to have the clock already enabled if a crypto request arrives.